internal/network/nsx_subresource.go

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package network
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+// NsxSchema this helper function extracts the NSX schema, which
+// contains the parameters required to install and configure NSX in a workload domain.
+func NsxSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"vip": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Virtual IP address which would act as proxy/alias for NSX Managers",
+				ValidateFunc: validation_utils.ValidateIPv4AddressSchema,
+			},
+			"vip_fqdn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "FQDN for VIP so that common SSL certificates can be installed across all managers",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"license_key": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "NSX license value",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"form_factor": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "NSX manager form factor",
```

Comment:
```suggestion
				Description:  "Form factor for the NSX Manager appliance",
```

Should this have a validate of allowed values?

For example... 

```go
ValidateFunc: validation.StringInSlice(NsxManagerFormFactorAllowedValues, true),
```